sdk/metric/manual_reader.go

```diff
@@ -0,0 +1,62 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metric // import "go.opentelemetry.io/otel/sdk/metric/reader"
+
+import (
+	"context"
+	"fmt"
+
+	"go.opentelemetry.io/otel/sdk/metric/export"
+)
+
+// ManualReader is a a simple Reader that allows an application to
+// read metrics on demand.  It simply stores the Producer interface
+// provided through registration.  Flush and Shutdown are no-ops.
+type ManualReader struct {
```

Comment:
I think we have gotten to an impasse on "Just Design" and I think either an implementation to demonstrate how this would work, or some logic in the MetricProducer of how this is used is warranted. 

I for one have already seen the use case of the `Shutdown` method and will have to work to add that code into this.  

But I still don't see how `ForceFlush` works in these implementations.  This leads me to believe that I don't think we should be designing these in the absence of some use case, either the minimal implementation or where it is used.

Reply:
I'd like to understand this concern about Shutdown and ForceFlush. They way I understand both of them, both method calls pass-through to each of the readers (i.e., exporters). If a reader (i.e., exporter) has nothing applicable to do for flush or shutdown, they'll do nothing. The reason we want Reader, Producer, and the Produce() method to be public, is so that users and/or exporters can choose to perform manual reading simply by implementing the 1-line `Register` function plus custom Shutdown and ForceFlush logic.